### ISSUE TITLE
weight_decay in Adam is not an L2 Penalty

### ISSUE THREAD

**Issue by RemiDav** [pytorch/pytorch#38767]
## 📚 Documentation

The documentation for torch.optim.Adam states:
weight_decay (float, optional): weight decay (L2 penalty) (default: 0)

The formula for weight decay and L2-regularization are different.
Weight decay:
w[t+1] = w[t] - learning_rate * dw - weight_decay * w

L2-regularization:
loss = actual_loss + lambda * 1/2 sum(||w||^2)

For SGD, a weight decay is similar to an L2 regularization, which is not true for adaptive gradient.
See "Decoupled Weight Decay Regularization" ( https://arxiv.org/abs/1711.05101 )

Which version is implemented in PyTorch ? Weight decay ? L2-regularization ?
A formula in the documentation would clarify that.

**Comment by Sanghyuk-Choi** [comment]
'weight_decay' is different from 'learning_rate_decay'.
In Pytorch, learning_rate_decay used for w[t+1] = w[t] - learning_rate * dw - **learning_rate_decay** * w.
loss = actual_loss + lambda * 1/2 sum(||w||^2) -> weight_decay= lambda

**Comment by dvolgyes** [comment]
No offense, but it is very badly documented.
This issue has been raised several times, e.g. #50464   #42843  #43055
The Adam algorithm implements the L2 regularization,
and for long time it implied otherwise. 
Now it is just short on documentation, 
but at least it does not imply implementing the decoupled weight decay.

But I think the documentation should point to AdamW and explain in a sentence
that L2 regularization with momentum found to be suboptimal,
see the Decoupled Weight Decay paper. 

Anyhow, either the documentation should be clarified further, or just the open tickets should be closed, there are at least the above two open and this.
